setupini.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+
+###This shell script is used to setup initial environment during login. This checks whether a psql database is up or not. Or,
+### installed or not. If not, it will start the database or create a new one
+### Checks recent python version. If not upgraded, it upgrades it and set it as default alternative
+### Install required dependencies from requirements.txt
+
+### Database installation check
+
+db_exist=`which psql | wc -l`
+echo $db_exist
+
+if [ $db_exist -eq 1 ]; then
+        echo "Database is installed. Checking its status..."
+        db_status=$(sudo service postgresql status)
+        db_check=$(echo $db_status | grep -i "stopped"|wc -l)
+        if [ $db_check -eq 0 ]; then
+                echo "Database is running. No need to start ...."
+        else
+                echo "Database is not running. Starting the database...."
+                start_db=`sudo service postgresql start`
+                if [ $? -eq 0 ]; then
+                        echo "Database started"
+                fi
+        fi
+else
+        sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
+        if [ $? -eq 0 ]; then
+                wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+
+        fi
+        if [ $? -eq 0 ]; then
+                echo "Key added"
+                sudo apt-get update
+        fi
+        if [ $? -eq 0 ]; then
+                echo "Updates of the packages done."
+                sudo apt-get install postgresql
+        fi
+        if [ $? -eq 0 ]; then
+                echo "PostGRE installed. Checking version"
+                version=`psql --version`
+                echo $version
+        fi
+
+
+fi
+
+
+### Python installation check
+
+echo "Checking python installation"
+py_status=$(python3 --version | wc -l)
+py_version=$(python3 --version)
+py_ver=$(echo $py_version | cut -d " " -f2)
+if [ $py_status -ge 1 ]; then
+        echo "Python is installed and version is $py_version"
+        if [$py_ver = *Python 3.9* ]; then
+                echo "Python 3.9+ installed. No further installation needed"
+        else
+                echo "Python 3.9 is needed"
+                sudo apt update
```

Comment:
I would get rid of this (and the `sudo apt-get update` on line 35) and do `sudo apt update` as the first thing when the script runs. No need to ever do it twice.